synapse/api/ratelimiting.py

```diff
@@ -93,6 +98,8 @@ def can_do_action(
         """Can the entity (e.g. user or IP address) perform the action?
 
         Args:
+            requester: The requester that is doing the action, if any. Used to check for
+                ratelimit overrides.
```

Comment:
I think it's worth being explicit here that we only check for the presence of a non-zero override.